carbon-design-system/carbon-addons-iot-react
# feat(table): add column anticipation to multi-sort clicks

Closes #2398 

**Summary**

- Adds the column that the user clicked "multi-sort" from into the modal UI, to save them the extra step of adding it manually.

**Change List (commits, features, bugs, etc)**

- add `anticipatedMultiSortColumn` prop
- adds tests to confirm
- fixes missing defaultProps for multisort

**Acceptance Test (how to verify the PR)**

- [StatefulTable with multi-sort](https://deploy-preview-2994--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-table-statefultable--with-multi-sorting)
- click an unsorted column 3-dot menu
- click "Multi-sort"
- the modal should open with the column you selected already chosen
- cancel or close the model and open from a different column
- the new column should be chosen
- Click "sort"
- The new column should show as sorted

Run through the various combinations of clearing, adding, etc. to confirm.

**Regression Test (how to make sure this PR doesn't break old functionality)**

- multi-sort still works as expected



**Things to look for during review**
